lib/socketpair.h

```diff
@@ -26,19 +26,35 @@
 
 #include "curl_setup.h"
 
-#ifdef HAVE_PIPE
+#if defined(HAVE_EVENTFD) && \
+    defined(__x86_64__) && \
+    defined(__aarch64__) && \
+    defined(__ia64__) && \
+    defined(__ppc64__) && \
+    defined(__mips64) && \
+    defined(__sparc64__) && \
+    defined(__riscv_64e) && \
+    defined(__s390x__)
+
+/* Use eventfd only on 64-bit CPU architectures */
```

Comment:
I'm not against this but I'm curious: why 64 bit only?

Reply:
`eventfd` has a stringent policy of requiring the 8-byte buffer when calling `read(2)` and `write(2)` on it. In some rare cases, the C standard library implementation on a 32-bit system might choose to define uint64_t as a 32-bit type for various reasons (memory limitations, compatibility with older code), which makes `eventfd` broken.

Ref: https://github.com/curl/curl/actions/runs/9362009840/job/25769979500